src/lib/yaraparse.py

```diff
@@ -153,20 +159,114 @@ def ursify_nocase_bytes(raw: bytes) -> UrsaExpression:
     return UrsaExpression(f"{{{ ' '.join(out) }}}")
 
 
-def ursify_plain_string(string: PlainString) -> UrsaExpression:
-    text_ascii = string.pure_text
-    text_wide = bytes(x for y in text_ascii for x in [y, 0])
+def encode_wide_bytes(raw: bytes) -> bytes:
+    return bytes(x for y in raw for x in [y, 0])
+
+
+def flatten_regex_or_tree(unit: Any) -> Optional[List[bytes]]:
+    if type(unit) is RegexpText:
+        return [unit.text.encode()]
+    elif type(unit) is RegexpOr:
+        left = flatten_regex_or_tree(unit.left)
+        right = flatten_regex_or_tree(unit.right)
+        if not left or not right:
+            return None
+        return left + right
+    elif type(unit) is RegexpConcat:
+        chars = [flatten_regex_or_tree(u) for u in unit.units]
+        string = b""
+        for c in chars:
+            if c is None:
+                return None
+            string += c[0]
+        return [string]
+    else:
+        return None
+
+
+def urisfy_regex(
+    units: List[Any],
+    is_ascii: bool = False,
+    is_wide: bool = False,
+    is_nocase: bool = False,
+) -> Optional[UrsaExpression]:
+    strings: List[UrsaExpression] = []
+
+    joined_string = b""
+    for i, unit in enumerate(units):
+        if type(unit) is RegexpText:
+            joined_string += unit.text.encode()
+        elif type(unit) is RegexpGroup:
+            or_strings = flatten_regex_or_tree(unit.unit)
+            if or_strings and all(s is not None for s in or_strings):
+                or_ursa_strings = [
+                    ursify_plain_string(
+                        s,
+                        is_ascii=is_ascii,
+                        is_wide=is_wide,
+                        is_nocase=is_nocase,
+                    )
+                    for s in or_strings
+                ]
+                strings.append(UrsaExpression.or_(*or_ursa_strings))
+
+        if joined_string and (
```

Comment:
why the `joined_string`? As far as I can see, it's only non-empty when `if type(unit) is RegexpText`, and then it's immediately changed back to `b""`. Maybe this if can be moved inside the `if type(unit) is RegexpText` branch?

Reply:
it's not immediately changed, it's used to collect the single characters and separate them when another type of node occurs